atlassian/github-for-jira
# Arc-1453 select GH version view to GHE server url view

**Rationale:**
Continuing task of wiring up the UI screens. This PR focuses on navigating from the GitHub version selection screen (cloud/server):

- if cloud is selected, user is redirected along current install flow (logging into GH then the connect an org screen)
- if server is selected, user is redirected to the GHE server url screen. They can also navigate back to the previous screen.

**Changes:**

- Added a new `JiraServerUrlRouter` and middleware `JiraServerUrlGet` to handle rendering the GHE server url screen and navigation
- updated functionality in static/js/jira-select-card-option.js so we can handle navigating users down the cloud/server path.
- renamed jira-connect-ghe.hbs/.css jira-server-url.hbs/.css. We're not really connecting to ghe here so thought it would be better if the file files were closer to what the view is for (inputing the server url)
- added a new module to connect
- updated snapshots for our connect object
- **What's not included**: styling clean-up. There's a noticeable size difference in the containers between these 2 views but we can clean this up later on.

**Demo:**
Like the last PR, the new path is hidden behind a FF.

With FF set to false we still navigate from the config screen to the connect an org screen:

https://user-images.githubusercontent.com/37155488/176094892-a9aea163-eca0-493c-8365-111f24b6ae07.mov

With the FF set to true (done manually on my local for demo purposes only) we can navigate from the config to the GH select version page. From here we can:

- Select the cloud option which then navigates us along the current path.

https://user-images.githubusercontent.com/37155488/176096059-78c7f61f-9a24-43d0-a173-5020fd056539.mov

- Select the server option which then navigates us along the new ghe server path, specifically to the GHE server url screen. We can also navigate back to the previous screen.

https://user-images.githubusercontent.com/37155488/176096266-885a32f8-aacf-4c32-bbaa-c131521d3ef1.mov

**To test locally:**
In src/routes/jira/configuration/jira-configuration-get.ts change the value from false to true for the FF check for handleNavigationClassName. After this PR is merged and deployed to prod, I'll turn the FF on for our segment so we can start testing these changes in the prod app. **NOTE:** you will need to uninstall and reinstall the app for Jira to detect the new modules.

**What's up next:**
In [Arc 1455](https://softwareteams.atlassian.net/jira/software/projects/ARC/boards/385?assignee=604894ffe19f910069497c6a&issueParent=125881&selectedIssue=ARC-1455) I'll add navigation from the GHE Server URL view 

<img width="905" alt="Screen Shot 2022-06-28 at 4 26 09 pm" src="https://user-images.githubusercontent.com/37155488/176108235-535d2daf-7324-4c4a-a56f-941122e0e1c7.png">

to the manual vs auto app creation page.

<img width="716" alt="Screen Shot 2022-06-28 at 4 26 23 pm" src="https://user-images.githubusercontent.com/37155488/176108192-591008e8-2e72-428a-94dd-3080efe5dfce.png">

Please see the [description and checklist](https://softwareteams.atlassian.net/jira/software/projects/ARC/boards/385?assignee=604894ffe19f910069497c6a&issueParent=125881&selectedIssue=ARC-1455) for what will and won't be included in that PR.
